grr/server/grr_response_server/gui/ui/store/client_page_facade.ts

```diff
@@ -355,9 +356,18 @@ export class ClientPageStore extends ComponentStore<ClientPageState> {
   /** An effect querying results of a given flow. */
   private readonly queryFlowResultsImpl = this.effect<FlowResultsQuery>(
       obs$ => obs$.pipe(
-          takeUntil(this.selectedClientIdChanged$),
-          withLatestFrom(this.selectedClientId$),
-          exhaustMap(([query, clientId]) => {
+        takeUntil(this.selectedClientIdChanged$),
+        withLatestFrom(this.selectedClientId$),
+        // Below we are grouping the requests by flowId, tag and type, and
+        // for each group we use a queuedExhaustMap with queue size 1 to send http request
+        // for results.
+        // TODO: Think about how we should split the code so it's cleaner
+        // TODO: Add tests to ensure the intended functionality
+        groupBy(([query, clientId]) => {
+          return [query.flowId, query.withTag, query.withType].join('-');
```

Comment:
Separate this into a function.

Also, URI-encode the characters and use a special delimiter which is encoded to a different value.

Reply:
Also provide a small test for the function.